product_pricelist_brand_required/models/product_pricelist.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2025 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ProductPricelist(models.Model):
+    _inherit = "product.pricelist"
+
+    def _get_applicable_rules_domain(self, products, date, **kwargs):
+        return super()._get_applicable_rules_domain(products, date, **kwargs) + [
```

Comment:
Wouldn't this always cause this domain to be searched?

Reply:
Yes, and that's what we want